### PR TITLE
Accept FSim(pi/2, pi/6) in ConvertToSycamoreGates

### PR DESCRIPTION
**Pull request by maffoo**
Currently the optimizer accepts the `SycamoreGate` specifically, but fails on an equivalent `FSimGate`. This adds support for fsim gates with sycamore parameters (theta = pi/2, phi = ph/6).

Also adds support for PhasedXZGate as a native gate that will be passed through when optimizing for sycamore.

### CODE REVIEW

**Comment by maffoo** [comment]
Incidentally, gate pattern matching would be really helpful for situations like this (see #2536).

File: cirq/google/optimizers/convert_to_sycamore_gates.py [hunk 1]
```diff
@@ -75,10 +75,20 @@ def _is_native_sycamore_op(self, op: ops.Operation) -> bool:
...
         """
-        return (isinstance(op, ops.GateOperation) and isinstance(
-            cast(ops.GateOperation, op).gate,
-            (SycamoreGate, ops.MeasurementGate, ops.PhasedXPowGate,
-             ops.XPowGate, ops.YPowGate, ops.ZPowGate)))
+        gate = op.gate
+
+        if isinstance(
+                gate,
+            (SycamoreGate, ops.MeasurementGate, ops.PhasedXZGate,
```
Comment (dstrain115) [inline cirq/google/optimizers/convert_to_sycamore_gates.py]: Maybe add a quick note that this also adds the PhasedXZGate?

File: cirq/google/optimizers/convert_to_sycamore_gates_test.py [hunk 2]
```diff
@@ -38,6 +38,16 @@ def test_convert_to_sycamore_gates_swap_zz():
...
+def test_convert_to_sycamore_gates_fsim():
+    q0, q1 = cirq.LineQubit.range(2)
+    circuit = cirq.Circuit(
+        cirq.FSimGate(theta=np.pi / 2, phi=np.pi / 6)(q0, q1))
+    compiled_circuit = circuit.copy()
+    cgoc.ConvertToSycamoreGates()(compiled_circuit)
+
+    cirq.testing.assert_same_circuits(circuit, compiled_circuit)
+
+
```
Comment (dstrain115) [inline cirq/google/optimizers/convert_to_sycamore_gates_test.py]: Worth also adding a test for PhasedXZ?

Comment (maffoo) [reply cirq/google/optimizers/convert_to_sycamore_gates_test.py]: Added a test.

File: cirq/google/optimizers/convert_to_sycamore_gates.py (hunk 1 above)
Comment (maffoo) [reply cirq/google/optimizers/convert_to_sycamore_gates.py]: Will note in the commit message.

**Pull request merged**